### ISSUE TITLE
Generic object literal completions include any random member you’re currently typing

### ISSUE THREAD

**Issue by mjbvz** [microsoft/TypeScript#35268]
From https://github.com/microsoft/vscode/issues/85185

**TypeScript Version:**  3.8.0-dev.20191121

**Search Terms:**
- suggest / suggestions / completions / intellisense
- optional
- extends

**Code**
For the code:

```ts
interface Test {
    keyPath?: string;
    autoIncrement?: boolean;
}

function test<T extends Record<string, Test>>(opt: T) { }

test({
    a: {
        keyPath: '',
        a
    }
})
```

Trigger suggestions after `a` in the call to `test`.

**Expected behavior:**
Suggestion for  `autoIncrement` returned.

**Actual behavior:**
No suggestion for `autoIncrement`. Only a single suggestion for `a` is returned

```
[Trace  - 18:58:01.966] <semantic> Sending request: completionEntryDetails (90). Response expected: yes. Current queue length: 0
Arguments: {
    "file": "/Users/matb/projects/san/a.ts",
    "line": 11,
    "offset": 10,
    "entryNames": [
        "a"
    ]
}
```

**Comment by RyanCavanaugh** [comment]
@andrewbranch I think you have a duplicate of this you've been looking at?

**Comment by andrewbranch** [comment]
This is mostly solved by #34855, except the bogus `a` completion is still there. I think @DanielRosenwasser told me we attempt to filter out object literal completions that are exactly what you’ve already typed, but I’ve noticed them come up basically all the time, so either that filtering is very broken or does not indeed exist.
